src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include <UIAutomationCore.h>
+#include "TermControlAutomationPeer.h"
+#include "TermControl.h"
+#include "TermControlAutomationPeer.g.cpp"
+
+#include "XamlUiaTextRange.h"
+
+namespace winrt::Microsoft::Terminal::TerminalControl::implementation
+{
+    TermControlAutomationPeer::TermControlAutomationPeer(winrt::Microsoft::Terminal::TerminalControl::implementation::TermControl const& owner) :
+        TermControlAutomationPeerT<TermControlAutomationPeer>(owner),
+        _uiaProvider{ owner.GetRenderData(), nullptr, std::bind(&TermControlAutomationPeer::GetBoundingRectWrapped, this) } {};
+
+    winrt::hstring TermControlAutomationPeer::GetClassNameCore() const
+    {
+        return L"TerminalControl";
+    }
+
+    AutomationControlType TermControlAutomationPeer::GetAutomationControlTypeCore() const
+    {
+        return AutomationControlType::Text;
+    }
+
+    winrt::hstring TermControlAutomationPeer::GetLocalizedControlTypeCore() const
+    {
+        return L"TerminalControl";
+    }
+
+    winrt::Windows::Foundation::IInspectable TermControlAutomationPeer::GetPatternCore(PatternInterface patternInterface) const
+    {
+        switch (patternInterface)
+        {
+        case PatternInterface::Text:
+            return *this;
+            break;
+        default:
+            return nullptr;
+        }
+    }
+
+#pragma region ITextProvider
+    winrt::com_array<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider> TermControlAutomationPeer::GetSelection()
+    {
+        try

```

Comment:
do we actually want to catch exceptions here? C++/winrt will translate any exception into an HRESULT on exit, which may be good

Reply:
So, should I remove the big try-catch, and add specific logic for destroying the SAFE_ARRAY if something happens? (kinda like the logic in ScreenInfoUiaProvider's GetSelection function)

Reply:
I guess the same question about destroying the SAFE_ARRAY applies to a lot of the other functions too.

Reply:
If cppwinrt already has provisions to deal with exceptions across a call boundary, I think we leverage them.